src/converse-service-administration.js

```diff
@@ -0,0 +1,412 @@
+            },
+
+            render () {
+                this.el.innerHTML = tpl_service_admin_option({
+                    'name': this.model.get('command_name'),
+                    'service_admin_name': this.model.get('command_name')
+                });
+                return this.el;
+            },
+
+            openServiceAdminChat () {
+                const stanza = $iq({
+                    'from': _converse.connection.jid,
+                    'id': u.getUniqueId(),
+                    'to': _converse.domain,
+                    'type': 'set',
+                    'xml:lang': _converse.locale
+                }).c('command', {
+                    xmlns: Strophe.NS.COMMANDS,
+                    action: 'execute',
+                    node: this.model.attributes.command_node
+                }).up();
+
+                _converse.connection.sendIQ(stanza, result => {
+                    const session_id = sizzle('command', result).pop().getAttribute('sessionid');
+                    const fields = sizzle('field', result);
+                    const form_type_value = sizzle('field[var="FORM_TYPE"] > value', result).pop().innerHTML;
+
+                    const attributes = {
+                        'id': _converse.SERVICE_ADMIN_TYPE,
+                        'jid': _converse.domain,
+                        'type': _converse.SERVICE_ADMIN_TYPE,
+                        'from': _converse.domain,
+                        'fields': Array.from(fields,
+                                             field => this.getAttributesFromField(field)),
+                        'form_type_value': form_type_value,
+                        'name': this.model.get('name'),
+                        'node': this.model.get('command_node'),
+                        'header_text': this.model.attributes.description,
+                        'send_button_text': __('send'),
+                        'placeholder_text': this.model.attributes.description,
+                        'session_id': session_id
+                    };
+
+                    const existing_chatbox = _converse.chatboxes.get(_converse.SERVICE_ADMIN_TYPE);
+                    if (existing_chatbox) {
+                        existing_chatbox.set(attributes);
+                        existing_chatbox.maybeShow(true);
+                    } else {
+                        const chatbox = new _converse.ServiceAdminBox(attributes);
+
+                        _converse.chatboxes.add(chatbox);
+                        chatbox.maybeShow(true);
+                    }
+                }, error => {
+                    _converse.log(error, Strophe.LogLevel.ERROR);
+                });
+            },
+
+            getAttributesFromField (field) {
+                const attribute_names = field.getAttributeNames();
+                const obj = {};
+                attribute_names.forEach(name => {
+                    obj[name] = field.getAttribute(name);
+                });
```

Comment:
Not really important, but I like using `reduce` instead of `forEach` to create an object from an array.
You can usually save a few lines of code that way.

For example, you can replace the above 3 lines with one line:
```
const obj = attribute_names.reduce((o, n) => (o[n] = field.getAttribute(n)), {});
```

Reply:
I belive this won't work because `(o[n] = field.getAttribute(n))` returns the assigned value and not the complete object.